### PR TITLE
Fixes/#1098 mastr units egon2021 commissioning date

### LINKED ISSUE #1098: Units from MaStR dataset are not filtered by commissioning date
The new MaStR dataset added in #1063 contains all data up to 2022-11-17 which is ok if filtered in subsequent power plant tasks (which does not happen except for PV rooftop). This is not suitable for the `eGon2021` scenario.
I'll add 2021-12-31 as upper limit for the commissioning date during import.

@ulfmueller @khelfen

### PR DESCRIPTION
**Pull request by nesnoj**
Fixes #1098 

## Before merging into `dev`-branch, please make sure that

- [x] the `CHANGELOG.rst` was updated.
- [x] new and adjusted code is formated using `black` and `isort`.
- [x] the `Dataset`-version is updated when existing datasets are adjusted.
- [x] the branch was merged into the
      `continuous-integration/run-everything-over-the-weekend`-branch.
- [ ] the workflow is running successful in `test mode`.
- [x] the workflow is running successful in `Everything` mode.

### CODE REVIEW

**Review by khelfen — APPROVED** [review]
I didn't test it, but the changes look fine :)

File: src/egon/data/datasets/power_plants/mastr.py [hunk 1]
```diff
@@ -313,6 +313,23 @@ def voltage_levels(p: float) -> int:
         units = units.loc[units.Land == "Deutschland"]
         print(f"    {len_old-len(units)} units outside of Germany dropped...")
 
+        # drop units installed after reference date (eGon2021 scenario)
+        len_old = len(units)
+        ts = pd.Timestamp("2022-01-01 00:00:00")
```
Comment (khelfen) [inline src/egon/data/datasets/power_plants/mastr.py:318]: Except maybe move `"2022-01-01 00:00:00"` into the `datasets.yml`?

Comment (nesnoj) [reply src/egon/data/datasets/power_plants/mastr.py]: Good point, done.

**Comment by nesnoj** [comment]
> I didn't test it, but the changes look fine :)

Thx!

**Comment by khelfen** [comment]
@nesnoj should I merge this or is there something that still needs to be done?

**Comment by nesnoj** [comment]
> @nesnoj should I merge this or is there something that still needs to be done?

I quickly checked the tables in  `supply.egon_power_plants_*` in the current run and they look good.
Only the PV roof one `supply.egon_power_plants_pv_roof_building` has not been created yet. But you get your data from `supply.egon_power_plants_pv`, right? If so, you can go ahead and merge :)

**Pull request merged**